src/main/java/spleetwaise/transaction/logic/parser/ParserUtil.java

```diff
@@ -61,6 +63,33 @@ public static Date parseDate(String date) throws ParseException {
         return new Date(date);
     }
 
+    /**
+     * Parses a {@code String indexStr} into a {@code int}. Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code indexStr} is invalid.
+     */
+    public static int parseIndex(String indexStr) throws ParseException {
+        requireNonNull(indexStr);
+        indexStr = indexStr.trim();
+        int index;
+        try {
+            index = Integer.parseInt(indexStr);
+        } catch (NumberFormatException e) {
+            throw new ParseException("Index is not in integer form");
+        }
+
+        return index;
+    }
+
+    /**
+     * Parses a {@code String catStr} into a {@code String} that represents the category.
+     * Leading and trailing whitespaces will be trimmed and capitalized
+     */
+    public static String parseCategory(String catStr) {
+        requireNonNull(catStr);
+        return catStr.trim().toUpperCase();
```

Comment:
Referencing other parse methods, usually, user input validation and sanitisation are done here as well. It looks like you have missed out on checking if the category string is empty here.